### DISCUSSION TITLE
Add a way to get searchParams in Server Actions

### DISCUSSION THREAD

**Discussion by ptvty** [vercel/next.js#74156] [App Router] [answered]
Related: https://github.com/vercel/next.js/issues/50156
Looking at the Network tab I can see that the POST request sent to the server action on form submit already includes all the search params from the current URL, but I couldn't find a way to get them from within the server action function.

There are valid use cases where this is needed.

I'm using a server action as the form action in a pure server component.
The URL already includes multiple search params and I need these params in addition to the form data.
So currently for the server action to have these params I had to get `searchParams` in the server component, and pass them using an `<input type="hidden" />` so I can get those in the handler function as the form data.

This is obviously not efficient as the same data is present in both the URL and the form data!

**Comment by tochizawa** [comment, marked as answer]
When using values other than FormData in Server Actions, consider using `bind`.
However, since bind implicitly generates `<input type="hidden" />`, it may not be a fundamental solution as it is essentially the same as manually creating the hidden input.
Please check the following link and sample code for reference.

```
async function myAction(searchParams: { [key: string]: string | string[] | undefined }, formData: FormData) {
  "use server";
  ...
}

export default async function Page(props: {
  searchParams: Promise<{ [key: string]: string | string[] | undefined }>;
}) {
  const searchParams = await props.searchParams;
  const actionWithSearchParams = actiona.bind(null, searchParams);
  return (
    <form action={updateUserWithId}>
      <button type="submit">submit</button>
    </form>
  );
}
```

https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#passing-additional-arguments
